lib/api/socket-handler.js

```diff
@@ -31,6 +32,15 @@ var SocketHandler = function(app, server, auth) {
   var self = this;
   this.io.sockets.on('connection', function(socket) {
     var clientQuery;
+    
+    // override the emit function to add logging
+    var emit = socket.emit;
+    socket.emit = function(event, data) {
+      logger.debug(event, data);
+      
+      socket.emit = emit;
+      socket.emit(event, data);
```

Comment:
I dunno, it still seems to me that since `emit` holds a reference to the original `socket.emit`, the first time the new `socket.emit` is called, it will reassign the original method to `socket.emit`, and the new `socket.emit` will get garbage collected since nothing will refer to it anymore. So the new `socket.emit` will only ever get called once per connection. If you replace lines 41-42 with just `emit(event, data)`, that won't happen.